### PR TITLE
refactor dump_failures

### PR DESCRIPTION
**Pull request by carcablop**
refactor dump_failures script to have an excluded manifest

### CODE REVIEW

File: citools/dump_failures.py [hunk 1]
```diff
@@ -19,6 +19,14 @@
...
     lambda: defaultdict(lambda: defaultdict(lambda: [0, 0]))
 )
+SKIPPED_MANIFESTS = []
+
+
+def red_file_ini():
+    file_config = configparser.configparser()
+    file_config.read("dom/canvas/test/mochitest.ini")
+    for test in file_config.sections():
+        SKIPPED_MANIFESTS.append(test)
```
Comment (jmaher) [inline citools/dump_failures.py:29]: great start on this.  I think you went too far here- it is ok a great opportunity for me to explain manifest vs test.  We have hundreds of manifest files (.ini) in the mozilla-central source tree.  Each manifest file contains a set of tests (between 1 and 400+, most contain 20-60).  For SKIPPED_MANIFESTS we want to skip manifests that have special cases.  We don't care as much about the individual tests.  While you did a great job of reading the manifest and adding each test to the SKIPPED_MANIFEST, that isn't needed here, just a list of .ini files.

Comment (carcablop) [reply citools/dump_failures.py:29]: Ok Thanks, now I see it and understand it better. "mochitest.ini" is the manifest file, which I must add to the list.

Comment (carcablop) [reply citools/dump_failures.py:29]: Hello. I have implementd this piece of code
"def update_skipped_manifest():
    path_list = pathlib.Path("dom/canvas/test").glob("**/*.ini")
    for skip_manifest in path_list:
        if skip_manifest.name == "mochitest.ini":
            SKIPPED_MANIFESTS.append(skip_manifest)"
This allows me to find all the .ini files that are in that directory, in this case that path is from the mozilla-central source tree, but in this project (mozci-tools) my sistem couldn't access it. What else should I take into account to go through this directory and not generate failures?. Anyway I executing the test, but isort failed. Could this failure be due to the above mentioned?.

**Comment by jmaher** [comment]
Did you have any remaining questions?  I would like to make sure your work gets merged in and you can keep moving forward.

File: citools/dump_failures.py [hunk 2]
```diff
@@ -20,14 +20,22 @@
...
+        # make absolute path in case this has symlinks
+        path_absolute = pathlib.Path(path).resolve(strict=True).parent
+        SKIPPED_MANIFESTS.append(
+            found_manifest
+            for found_manifest in pathlib.Path(path_absolute).glob("**/*.ini")
+        )
+    except FileNotFoundError:
+        SKIPPED_MANIFESTS.append(path)
+    except RuntimeError:
+        SKIPPED_MANIFESTS.append(path)
```
Comment (jmaher) [inline citools/dump_failures.py:38]: where do you call find_related_manifest("dom/canvas/test") ?  I want to make sure we don't lose our reference to a known manifest we need to skip

Comment (carcablop) [reply citools/dump_failures.py:38]: I'm implementing the units tests, where I call the function find_related_manifest(path), with the path of the example "dom/canvas/test". I'm trying to mock the "pathlib.path.resolve()" function to generate the exception FileNotFoundError.

Comment (jmaher) [reply citools/dump_failures.py:38]: it sounds like you are handling this correctly, I appreciate the good coding practices.  We do need in dump failures to make sure that SKIPPED_MANIFESTS has `dom/canvas/test/mochitest.ini`

**Review by jmaher — APPROVED** [review]
great!

**Pull request merged**